src/test/java/org/breedinginsight/api/v1/controller/TraitUploadControllerIntegrationTest.java

```diff
@@ -78,11 +81,55 @@ public void setup() {
         // Insert user into program as not active
         dsl.execute(fp.get("InsertInactiveProgramUser"));
 
+        // Insert Trait
+        dsl.execute(fp.get("InsertProgramObservationLevel"));
+        dsl.execute(fp.get("InsertProgramOntology"));
+        dsl.execute(fp.get("InsertMethod"));
+        dsl.execute(fp.get("InsertScale"));
+        dsl.execute(fp.get("InsertTrait"));
+
         // Retrieve our new data
         validProgram = programDao.findAll().get(0);
 
     }
 
+    @Test
+    @Order(1)
+    void putTraitUploadDuplicatesInDB() {
```

Comment:
Do we have a card for being able to communicate back the rows that are duplicates for highlighting on the UI as specified in the design reference?

Reply:
https://trello.com/c/cF12yK1j/81-ont-81-report-a-count-of-imported-new-and-duplicate-traits-to-the-ui